JellyfishClientSdk/Tests/JellyfishClientSdkTests/JellyfishClientSdkTests.swift

```diff
@@ -1,12 +1,115 @@
+import MembraneRTC
 import XCTest
+import Mockingbird
 
+@testable import Starscream
 @testable import JellyfishClientSdk
 
+
 final class JellyfishClientSdkTests: XCTestCase {
-    func testExample() throws {
-        // This is an example of a functional test case.
-        // Use XCTAssert and related functions to verify your tests produce the correct
-        // results.
-        XCTAssertEqual(JellyfishClientSdk().text, "Hello, World!")
+  let mockedWebSocket = mock(JellyfishWebsocket.self)
+  let jellyfishClientListener = mock(JellyfishClientListener.self)
+  let testConfig = Config(websocketUrl: "ws:\\test.com", token: "testTOKEN")
+  var jellyfishClient: JellyfishClientInternal?
+  var webrtc: JellyfishMembraneRTC?
+  
+  func getMockWebsocket(url: String) -> JellyfishWebsocket {
+    return self.mockedWebSocket;
+  }
+  
+  static func generateDataFromMessage(_ message: Jellyfish_PeerMessage) -> Data {
+    guard let serializedData = try? message.serializedData() else {
+      return Data()
+    }
+    
+    return serializedData
+  }
+  
+  let authRequest = generateDataFromMessage(
+    Jellyfish_PeerMessage.with({
+      $0.authRequest = Jellyfish_PeerMessage.AuthRequest.with({
+        $0.token = "testTOKEN"
+      })
+    }))
+  
+  let authResponse = generateDataFromMessage(
+    Jellyfish_PeerMessage.with({
+      $0.authenticated = Jellyfish_PeerMessage.Authenticated()
+    }))
+  
+  let joinEvent = generateDataFromMessage(
+    Jellyfish_PeerMessage.with({
+      $0.mediaEvent = Jellyfish_PeerMessage.MediaEvent.with({
+        $0.data = "join"
+      })
+    })
+  )
+  
+  let sdpOfferEvent = generateDataFromMessage(
+    Jellyfish_PeerMessage.with({
+      $0.mediaEvent = Jellyfish_PeerMessage.MediaEvent.with({
+        $0.data = "sdpOffer"
+      })
+    })
+  )
+  
+  override func setUp() {
+    let webrtc = mock(JellyfishMembraneRTC.self)
+    let jellyfishClient = JellyfishClientInternal(listener: self.jellyfishClientListener, websocketFactory: getMockWebsocket)
+    jellyfishClient.webrtcClient = webrtc
+    self.jellyfishClient = jellyfishClient
+    self.webrtc = webrtc
+    
+    givenSwift(self.mockedWebSocket.connect()).will {
+      self.jellyfishClient?.websocketDidConnect()
     }
+  }
+  
+  func connect() {
+    jellyfishClient?.connect(config: self.testConfig)
+    verifyClientSent(authRequest)
+    sendToClient(authResponse)
+  }
+  
+  func sendToClient(_ data: Data) {
+    self.jellyfishClient?.websocketDidReceiveData(data: data)
+  }
+  
+  func verifyClientSent(_ data: Data) {
+    verify(self.mockedWebSocket.write(data: data)).wasCalled()
+  }
+  
+  func testConnectAndAuthenticate() throws {
+    connect()
+    verify(self.jellyfishClientListener.onAuthSuccess()).wasCalled()
+  }
+  
+  func testCleansUp() throws {
+    connect()
+    jellyfishClient?.cleanUp()
+    verify(self.mockedWebSocket.disconnect()).wasCalled()
+    verify(self.jellyfishClientListener.onDisconnected()).wasCalled()
+  }
```

Comment:
Now if we know how to verify that some webrtc method was called, what about adding `verify(self.webrtc?.disconnect()).wasCalled()` ?